content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md

```diff
@@ -360,6 +361,276 @@ GROUP BY _time, room
 {{% /expand %}}
 {{< /expand-wrapper >}}
 
+## date_bin_wallclock
+
+Calculates time intervals using the timezone of a specified time value and
+returns the start of the interval nearest to the specified timestamp.
+Use `date_bin_wallclock` to downsample time series data by grouping rows into
+time-based "bins" or "windows" that are based off "wall clock" times in a
+specific timezone and applying an aggregate or selector function to each window.
+
+### Time zone shifts
+
+Many regions use time zone shifts (such as Daylight Saving Time (DST) and British
+Summer Time (BST)). If a wall clock time bin starts at a time that does not
+exist in the specified time zone, the timestamp is adjusted to the time that is
+the same offset from the start of the day in that time zone.
+
+If a wall clock time represents an ambiguous time in the region then the
+behavior depends on the size of the specified interval. If the interval is
+larger than the difference between the two possible timestamps, then the earlier
+timestamp is used. Otherwise, the function uses the timestamp that matches the
+UTC offset of the input timestamp.
+
+```sql
+date_bin_wallclock(interval, expression[, origin])
+```
+
+##### Arguments:
+
+- **interval**: Bin interval.
+- **expression**: Time expression to operate on.
+  Can be a constant, column, or function.
+  The output timestamp uses the time zone from this time expression.
+- **origin_timestamp**: Starting point used to determine bin boundaries.
+  This must be a "wall clock" timestamp (no time zone).
+  _Default is the Unix epoch._
```

Comment:
It would be good to highlight a caveat:

Using an `origin` and `interval` such that there are bins within a discontinuity, but which do not align with the point at which the discontinuity occurs, can produce unexpected results. For example, considering the following query:
```
SELECT date_bin_wallclock(INTERVAL '1 hour', tz('2020-10-25T02:29:00+01:00', 'Europe/Paris'), '1970-01-01T00:30:00') AT TIME ZONE 'UTC';
```
The result is:
```
2020-10-24T23:30:00Z
``` 
However, the result of a nearly identical query:
```
SELECT date_bin_wallclock(INTERVAL '1 hour', tz('2020-10-25T02:30:00+01:00', 'Europe/Paris'), '1970-01-01T00:30:00') AT TIME ZONE 'UTC';
``` 
is:
```
2020-10-25T01:30:00Z
```
That is, timestamps only one minute apart are placed into bins that are two hours apart, despite the expected interval being one hour. Care should be taken to avoid such scenarios.

Reply:
In general advice to try and stay away from the discontinuities is probably sensible.

Reply:
So do avoid this, what would good recommendation be? Use an origin offset from the unix epoch that is greater than or equal to your bin interval?

Reply:
> Use an origin offset from the unix epoch that is greater than or equal to your bin interval?

Yes, either this^ or leave the `origin` as the default timestamp (e.g. `1970-01-01T00:00:00`).

I'd like to hear from @mhilton as well

Reply:
> Use an origin offset from the unix epoch that is greater than or equal to your bin interval?

Not exactly. My advice would generally be:

- If the bin interval is small (say <= 1 hour) `date_bin_wallclock` will probably do what one expects.
- If the interval is larger, a day would be typical, then make sure the origin time does not start any bin within a discontinuity. For pretty much all time zones leaving the default is fine which would start all the bins at midnight local time. There have been time zones that have skipped straight to 1am historically so this isn't completely safe.

I'm sorry I can't be more specific, but time zones are more of a policial thing than anything else so they can change at any point and with short notice (I've been burnt by that one before). Luckily The way most people will probably use `date_bin_wallclock`, in most parts of the world will work how they want.